raphtory/src/db/task/eval_vertex.rs

```diff
@@ -250,6 +310,93 @@ impl<'a, G: GraphViewOps, CS: ComputeState> EvalVertexView<'a, G, CS> {
             .read_with_pid(self.ss + 1, self.global_id(), self.pid(), agg_r)
             .unwrap_or(ACC::finish(&ACC::zero()))
     }
+
+    pub fn read_global_state_prev<A, IN, OUT, ACC: Accumulator<A, IN, OUT>>(
+        &self,
+        agg_r: &AccId<A, IN, OUT, ACC>,
+    ) -> OUT
+    where
+        A: StateType,
+        OUT: std::fmt::Debug,
+    {
+        self.global_state
+            .borrow()
+            .read_global(self.ss + 1, agg_r)
+            .unwrap_or(ACC::finish(&ACC::zero()))
+    }
+}
+
```

Comment:
Make ticket for migrating all of the results from algorithms into a `state` object which handles getting the results for a vertex given name/id/etc